src/displayapp/screens/FirmwareUpdate.cpp

```diff
@@ -78,4 +94,9 @@ void FirmwareUpdate::UpdateValidated() {
 void FirmwareUpdate::UpdateError() {
   lv_label_set_recolor(percentLabel, true);
   lv_label_set_text(percentLabel, "#ff0000 Error!#");
+  startTime = xTaskGetTickCount();
+}
+
+bool FirmwareUpdate::OnButtonPushed() {
```

Comment:
It is still possible to exit the app by swiping up and down. I think you can prevent that by overriding `bool Screen::OnTouchEvent(Pinetime::Applications::TouchEvents)`:

```
bool FirmwareUpdate::OnTouchEvent(Pinetime::Applications::TouchEvents) {
  return true;
}
```

Reply:
It shouldn't be possible to exit because the return gesture is set to `TouchEvents::None`, but maybe it actually processes the None event which allows the app to exit. I'd rather fix this by adding a new check before loading the return app which would be `returnTouchEvent != TouchEvents::None`

Reply:
Right! But when I tested it, it was still possible to close the app by swiping...
I think you need to add ` ReturnApp(Apps::Clock, FullRefreshDirections::Down, TouchEvents::None);` in the case `case Messages::BleFirmwareUpdateStarted:` of the eventloop of DisplayApp (line220), then.

Reply:
I added this check to TouchEvent message to make sure that TouchEvents::None isn't processed in any way. Before this I got the app to exit just before the update was finished, but couldn't get it after this.
```
if (gesture == TouchEvents::None) {
  break;
}
```

ReturnApp() shouldn't be needed in BleFirmwareUpdateStarted message because it already gets run inside LoadApp()

Reply:
You are right... Ok, I don't know what I did wrong... I'm sure I managed to close the update app with a swipe, but I cannot reproduce it anymore...

Reply:
Most likely the swipe gesture was what generated the TouchEvent message, but by the time displayapp read it, it had already changed to None in the touchscreen driver. The firmware update slows down processing, which probably added enough delay for this to happen.